src/platform/plugins/shared/discover/public/context_awareness/profile_providers/observability/metrics_data_source_profile/dummy_metrics_grid.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  EMPTY_OPTION,
+  ToolbarSelector,
+} from '@kbn/unified-histogram/components/chart/toolbar_selector';
+import type { ChartSectionContext } from '@kbn/unified-histogram/types';
+import React, { useCallback, useMemo } from 'react';
+
+// Dummy component to demonstrate how the actual metrics experience grid will hook into discover state and event handlers
+export const DummyMetricsGrid = ({
+  dataView,
+  renderToggleActions,
+  chartToolbarCss,
+  histogramCss,
+  services,
+  abortController,
+  onBrushEnd,
+  onFilter,
+  query,
+  searchSessionId,
+  timeRange,
+}: ChartSectionContext) => {
+  const [breakdownField, setBreakdown] = React.useState<DataViewField | undefined>(undefined);
+  const actions: IconButtonGroupProps['buttons'] = [
+    {
+      iconType: 'search',
+      label: i18n.translate('metricsExperience.searchButton', {
+        defaultMessage: 'Search',
+      }),
+
+      onClick: () => {},
+      'data-test-subj': 'metricsExperienceEditVisualization',
+    },
+    {
+      iconType: 'fullScreen',
+      label: i18n.translate('metricsExperience.fullScreenButton', {
+        defaultMessage: 'Full screen',
+      }),
+
+      onClick: () => {},
+      'data-test-subj': 'metricsExperienceEditVisualization',
+    },
+  ];
+
+  const fields = useMemo(() => dataView.fields.filter(fieldSupportsBreakdown), [dataView]);
+  const fieldOptions: SelectableEntry[] = useMemo(() => {
+    const options: SelectableEntry[] = fields
+      .map((field) => ({
+        key: field.name,
+        name: field.name,
+        label: field.displayName,
+        value: field.name,
+        checked:
+          breakdownField?.name === field.name
+            ? ('on' as EuiSelectableOption['checked'])
+            : undefined,
+        prepend: (
+          <span
+            css={css`
+              .euiToken {
+                vertical-align: middle;
+              }
+            `}
+          >
+            <FieldIcon {...getFieldIconProps(field)} />
+          </span>
+        ),
+      }))
+      .sort((a, b) => a.label.toLowerCase().localeCompare(b.label.toLowerCase()));
+
+    options.unshift({
+      key: EMPTY_OPTION,
+      value: EMPTY_OPTION,
+      label: i18n.translate('metricsExperience.breakdownFieldSelector.noBreakdownButtonLabel', {
+        defaultMessage: 'No breakdown',
+      }),
+      checked: !breakdownField ? ('on' as EuiSelectableOption['checked']) : undefined,
+    });
+
+    return options;
+  }, [fields, breakdownField]);
+
+  const onChange = useCallback<NonNullable<ToolbarSelectorProps['onChange']>>(
+    (chosenOption) => {
+      const newField = chosenOption?.value
+        ? fields.find((currentField) => currentField.name === chosenOption.value)
+        : undefined;
+      setBreakdown(newField);
+    },
+    [fields, setBreakdown]
+  );
+
+  // The structure here is VERY similar to that of the `chart.tsx` component
+  // Perhaps there should be a shared template for this?
```

Comment:
@jughosta wdyt about creating a common template?

Reply:
you mean inside UnifiedHistogram package for showing the toolbar and the rest? yes, that could be helpful.